pkg/ingester/ingester_lifecycle_test.go

```diff
@@ -268,3 +270,68 @@ func (i ingesterClientAdapater) TransferChunks(ctx context.Context, _ ...grpc.Ca
 func (i ingesterClientAdapater) Close() error {
 	return nil
 }
+
+func TestIngesterFlush(t *testing.T) {
+	cfg := defaultIngesterTestConfig()
+	store := newTestStore()
+
+	// Start the ingester, and get it into ACTIVE state.
+	ing, err := New(cfg, store)
+	require.NoError(t, err)
+
+	poll(t, 100*time.Millisecond, ring.ACTIVE, func() interface{} {
+		return ing.state
+	})
+
+	// Now write a sample to this ingester
+	var (
+		ts  = model.TimeFromUnix(123)
+		val = model.SampleValue(456)
+		m   = model.Metric{
+			model.MetricNameLabel: "foo",
+		}
+	)
+	ctx := user.Inject(context.Background(), userID)
+	_, err = ing.Push(ctx, util.ToWriteRequest([]model.Sample{
+		{
+			Metric:    m,
+			Timestamp: ts,
+			Value:     val,
+		},
+	}))
+	require.NoError(t, err)
+
+	// We add a 100ms sleep into the flush loop, such that we can reliably detect
```

Comment:
We are trying to test that two things happen sequentially and not in parallel, so I've slowed one of them down, waited for the other to happen, and then checked the slowed down one hasn't happened.

I can't really think of a better way to test for this that gives a negative signal when they happen in the wrong order.